geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync, unlinkSync } from 'fs';
+import { Extract } from 'unzipper';
+const mapshaper = require('mapshaper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          Extract({
+            path: `${fileInfo.destination}/${fileInfo.filename.replace(
+              '.zip',
+              '',
+            )}`,
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
```

Comment:
We were having a problem-related zip extraction, I guess this is part of the debugging work that Alex did.

Reply:
@kgajowy Actually the message is being logged by the logger in the public method. I added it as part of promisifying the stream in order to be able to `await` for the 'close' event of the stream and then call for other private method instead of using a callback in the same one